src/meta/src/controller/streaming_job.rs

```diff
@@ -565,7 +575,11 @@ impl CatalogController {
         txn.commit().await?;
 
         if !objs.is_empty() {
-            self.notify_frontend(Operation::Delete, build_relation_group(objs))
+            // We **may** also have notified the frontend about these objects,
+            // so we need to notify the frontend to delete them here.
+            // The frontend will ignore the request if the object does not exist,
+            // so it's safe to always notify.
+            self.notify_frontend(Operation::Delete, build_relation_group_for_delete(objs))
```

Comment:
As the notification now goes after successfully scheduling and building of the job, a failure here does not always mean that the notification was sent. That's why we changed all `DROP` handlers in the frontend to be `DROP .. IF EXISTS`.

An alternative could be introducing a new operation like `Cleanup` for this purposes, ensuring `Delete` remains strictly applied. But it seems too ad-hoc to me.

Reply:
Can there be a race condition? Whereby we have `Delete` of some catalog followed by `Add` of that catalog?

Reply:
I think not because `try_abort_creating_streaming_job` is called only after `create_streaming_job_inner` has returned with error, where `Add` is either pushed into the notification queue or not issued at all. After entering this `try_abort` function, no further `Add` operations will occur.

Reply:
🤔 We ensured that `CREATE` and `DROP` will be properly paired in #18476 .Even if the `CREATE` notification fails to send due to either a FE or meta node reboot or a broken connection between them, the creation of the mview and its internal table catalogs will still be synchronized through an initial snapshot.